prestart.js

```diff
@@ -1,43 +1,359 @@
+				removeActive(x);
+				if (currentFocus >= x.length) currentFocus = 0;
+				if (currentFocus < 0) currentFocus = (x.length - 1);
+				x[currentFocus].classList.add("autocomplete-active");
+				x[currentFocus].style['color'] = '#000000';
+				x[currentFocus].style['background-color'] = '#ffffff';
+			}
+		
+			function removeActive(x) {
+				/*a function to remove the "active" class from all autocomplete items:*/
+				for (let i = 0; i < x.length; i++) {
+					x[i].classList.remove("autocomplete-active");
+					x[i].style['color'] = '#ffffff';
+					x[i].style['background-color'] = '#00000000';
+				}
+			}
+		
+			function closeAllLists(elmnt) {
+				/*close all autocomplete lists in the document,
+				except the one passed as an argument:*/
+				let x = document.getElementsByClassName("autocomplete-items");
+						
+				for (let i = 0; i < x.length; i++) {
+					if (elmnt != x[i] && elmnt != inp) {
+						x[i].parentNode.removeChild(x[i]);
+					}
+				}
+			}
+			// Close list by clicking outside
+			document.addEventListener("click", function (e) {
+				closeAllLists(e.target);
+			});
+		}
+
+
+		/**
+		 * @description State state of the mod interface
+		 * @param {Modstate Constant} state 
+		 */
+		function setModState(state){
+
+			modState = state;
+
+			switch (state) {
+				case MOD_STATE.ALL_HIDDEN:
+					interfaceToggle.style.display = 'none';
+					modInterface.style.display = 'none';
+					break;
+				case MOD_STATE.TOGGLER_SHOWN:
+					interfaceToggle.style.display = 'block';
+					modInterface.style.display = 'none';
+					break;
+				case MOD_STATE.MOD_SHOWN:
+					interfaceToggle.style.display = 'none';
+					modInterface.style.display = 'block';
+					break;
+				default:
+					console.error('cc-teleport: Wrong mod state set');
+					break;
+			}
+
+			let x = document.getElementsByClassName("autocomplete-items");		
+			for (let i = 0; i < x.length; i++) {
+				x[i].parentNode.removeChild(x[i]);
+			}
+		}
+
+		// ----------------- Mod Injections --------------------------------------------------------------------
+
+		/**
+		 * @inject
+		 * Handle game state transitions from main menu to the game 
+		 */
+		sc.Combat.inject({
+			init(...args) {
+				this.parent(...args);
+				sc.Model.addObserver(sc.model, this);
+			},
+			modelChanged(model, event) {
+				if( model instanceof sc.GameModel ){
+					if(event == sc.GAME_MODEL_MSG.STATE_CHANGED || sc.GAME_MODEL_MSG.SUB_STATE_CHANGED ){
+						if( sc.model.isTitle() || !sc.model.isRunning())
+							setModState(MOD_STATE.ALL_HIDDEN);
+						if( sc.model.isGame() && sc.model.isRunning() )
+							setModState(MOD_STATE.TOGGLER_SHOWN);
+					}
+				}
+			}
+		});
+
+		/**
+		 * @inject
+		 * Add the interface to the mod 
+		 */
 		ig.Gui.inject({
 			init(...args) {
 				this.parent(...args);
+				
+				document.body.insertAdjacentHTML('beforeend',`
+					<img
+						id="interfaceToggle"
+						src="${'teleportIcon'.toPath(ig.root + 'mods/cc-teleport/', '.png')}" 
```

Comment:
You should probably create the `img` tag by hand (`new Image()` which returns `HTMLImageElement`, then set properties of the said image element etc) instead of passing a literal string into HTML, this opens path to XSS injections.